### DISCUSSION TITLE
Inserting a point outside the cuboid of a periodic 3D triangulation

### DISCUSSION THREAD

**Discussion by gy114** [CGAL/cgal#7926] [Q&A] [answered]
Hi, 

I'm doing some customized refinement of a periodic 3D Delaunay triangulation. Do we have an internal way to insert a periodic point outside the cuboid? I can write a small function to do that, but I was wondering if there is a more robust mechanism already implemented in CGAL. 

Thanks.

**Comment by MaelRL** [comment, marked as answer]
You can have a look at the function [robust_canonicalize_point](https://github.com/CGAL/cgal/blob/master/Periodic_3_triangulation_3/include/CGAL/Periodic_3_triangulation_3/internal/canonicalize_helper.h#L199).
